agreement/service_test.go

```diff
@@ -797,7 +797,7 @@ func setupAgreementWithValidator(t *testing.T, numNodes int, traceLevel traceLev
 
 	cleanupFn := func() {
 		for _, accessor := range dbAccessors {
-			defer accessor.Close()
+			accessor.Close()
```

Comment:
this one kept me puzzled for few seconds. The change I've made below was changing the close to work with a pointer receiver. That was causing this call to try and close the same accessor over and over since the variable `accessor` has a single memory location.